Bottom: faecc75884f52b705f942214b24af9369d87c033
Top:    9027537bce222f5156e4c694a506efc5d518e366
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-24 17:59:11 +0100

misc.c: Add query function IsPasteEmpty() to query a paste layer is empty


---

diff --git a/src/misc.c b/src/misc.c
index 02a883f..d63aef1 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -600,6 +600,22 @@ IsLayerGroupEmpty (int num)
   return true;
 }
 
+bool
+IsPasteEmpty (int side)
+{
+  bool paste_empty = true;
+  ALLPAD_LOOP (PCB->Data);
+  {
+    if (ON_SIDE (pad, side) && !TEST_FLAG (NOPASTEFLAG, pad) && pad->Mask > 0)
+      {
+        paste_empty = false;
+        break;
+      }
+  }
+  ENDALL_LOOP;
+  return paste_empty;
+}
+
 /* ---------------------------------------------------------------------------
  * gets minimum and maximum coordinates
  * returns NULL if layout is empty
diff --git a/src/misc.h b/src/misc.h
index b46cad4..cd7eb3d 100644
--- a/src/misc.h
+++ b/src/misc.h
@@ -47,6 +47,7 @@ bool IsDataEmpty (DataTypePtr);
 bool IsLayerEmpty (LayerTypePtr);
 bool IsLayerNumEmpty (int);
 bool IsLayerGroupEmpty (int);
+bool IsPasteEmpty (int);
 BoxTypePtr GetDataBoundingBox (DataTypePtr);
 void CenterDisplay (LocationType, LocationType, bool);
 void SetFontInfo (FontTypePtr);
